scripts/exregional_clean.ksh

```diff
@@ -1,58 +1,82 @@
 #!/bin/ksh --login
 
-currentime=`date`
-
+#
+#-----------------------------------------------------------------------
+# Source the variable definitions file.
+#-----------------------------------------------------------------------
+#
 . ${GLOBAL_VAR_DEFNS_FP}
+#
+#-----------------------------------------------------------------------
+# Save current shell options (in a global array).  Then set new options
+# for this script/function.
+#-----------------------------------------------------------------------
+#
+{ save_shell_opts; set -u -x; } > /dev/null 2>&1
+#
+#
+#-----------------------------------------------------------------------
+# set up currentime from CDATE 
+#-----------------------------------------------------------------------
+#
+currentime=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
 
+#-----------------------------------------------------------------------
 # Delete ptmp directories
-deletetime=`date +%Y%m%d -d "${currentime} 72 hours ago"`
+#-----------------------------------------------------------------------
+deletetime=$(date +%Y%m%d -d "${currentime} ${CLEAN_OLDPROD_HRS} hours ago")
 echo "Deleting ptmp directories before ${deletetime}..."
 cd ${COMOUT_BASEDIR}
-set -A XX `ls -d ${RUN}.20* | sort -r`
+set -A XX $(ls -d ${RUN}.20* | sort -r)
 for dir in ${XX[*]};do
-  onetime=`echo $dir | cut -d'.' -f2`
+  onetime=$(echo $dir | cut -d'.' -f2)
   if [[ ${onetime} -le ${deletetime} ]]; then
     rm -rf ${COMOUT_BASEDIR}/${RUN}.${onetime}
     echo "Deleted ${COMOUT_BASEDIR}/${RUN}.${onetime}"
   fi
 done
 
+#-----------------------------------------------------------------------
 # Delete stmp directories
-deletetime=`date +%Y%m%d%H -d "${currentime} 72 hours ago"`
+#-----------------------------------------------------------------------
+deletetime=$(date +%Y%m%d%H -d "${currentime} ${CLEAN_OLDRUN_HRS} hours ago")
 echo "Deleting stmp directories before ${deletetime}..."
 cd ${CYCLE_BASEDIR}
-set -A XX `ls -d 20* | sort -r`
+set -A XX $(ls -d 20* | sort -r)
 for onetime in ${XX[*]};do
   if [[ ${onetime} -le ${deletetime} ]]; then
```

Comment:
To be more robust, suggest change line 33, 47 and similar `if` statements to
`if [[ ${onetime} =~ '^[0-9]+$'  ]] && [[ ${onetime} -le ${deletetime} ]]; then
`
At the RAP/HRRR era, there was one time when someone changed, for example, `20190508` to `20190508.save` for debug purpose and then it crashed clean scripts for the next few days (maybe more than 10 days) , filled up the run disk very quickly.